src/main/scala/com/fulcrumgenomics/bam/Bams.scala

```diff
@@ -246,4 +248,34 @@ object Bams extends LazyLogging {
       }
     }
   }
+
+  /** Returns true if the read is mapped in an FR pair, false otherwise. */
+  def isFrPair(rec: SAMRecord): Boolean =
+    rec.getReadPairedFlag &&
+      !rec.getReadUnmappedFlag &&
+      !rec.getMateUnmappedFlag &&
+      rec.getReferenceIndex == rec.getMateReferenceIndex &&
+      SamPairUtil.getPairOrientation(rec) == PairOrientation.FR
+
+  /**
+    * Calculates the coordinates of the insert represented by this record and returns them
+    * as a pair of 1-based closed ended coordinated.
+    *
+    * Invalid to call on a read that is not mapped in a pair to the same chromosome as it's mate.
+    *
+    * @return the start and end position of the insert as a tuple, always with start <= end
+    */
+  def insertCoordinates(rec: SAMRecord): (Int, Int) = {
+    require(rec.getReadPairedFlag, s"Read ${rec.getReadName} is not paired, cannot calculate insert coordinates.")
+    require(!rec.getReadUnmappedFlag, s"Read ${rec.getReadName} must be mapped to calculate insert coordinates.")
+    require(!rec.getMateUnmappedFlag, s"Read ${rec.getReadName} must have mapped mate to calculate insert coordinates.")
+    require(rec.getReferenceIndex == rec.getMateReferenceIndex, s"Read ${rec.getReadName} must be mapped to same chrom as it's mate.")
+
+    val isize      = rec.getInferredInsertSize
+    val thisEnd    = if (rec.getReadNegativeStrandFlag) rec.getAlignmentEnd else rec.getAlignmentStart
```

Comment:
`thisEnd` was confusing since I thought it meant the `end position of this records` not what I think it means which is `the 5' coordinate of this record (end)`.